guides-and-tutorials/hello-stacks-quickstart-tutorial.md

```diff
@@ -1,652 +1,531 @@
 
-{% hint style="warning" %}
-If you run into funky rendering issues, be sure to change the network URL parameter in the address bar to `testnet`
-{% endhint %}
-
-Click on the little function symbol on the left and take the contract name you copied and paste it in the top field there, the name should be automatically populated.
-
-Hit 'Get Contract' and you should see the function we created there.
-
-Now we can click on that and type in a message to call the function. But if you try to do that now, you might get an error saying you don't have enough STX.
-
-How do we get STX tokens in our Devnet account? Remember that `Devnet.toml` file Clarinet generates? That determines what accounts have what tokens.
-
-So now we need to set up our Leather wallet extension that we set up at the beginning of this tutorial. If you already have an existing Leather wallet, you'll need to sign out in order to use it with the local Clarinet Devnet.
-
-{% hint style="danger" %}
-Make sure you copy your secret key before you sign out. You'll need it to restore your Leather wallet when you are done developing.
-{% endhint %}
+```clarity
+(define-public (add-message (content (string-utf8 280)))
+  (let ((id (+ (var-get message-count) u1)))
+    (map-set messages id content)
+    (map-set message-authors id tx-sender)
+    (var-set message-count id)
+    (ok id)))
+```
 
-Go ahead and copy the mnemonic listed in the `Devnet.toml` file and use that to import your local wallet into the Leather wallet. Then we can use that for interacting with our local Devnet chain.
+Let me walk you through this step by step:
 
-Once you do that, go ahead and restart `clarinet devnet start`.
+When you call [`define-public`](../reference/functions.md#define-public), you're creating a function that anyone can call from outside the contract. The function takes one parameter called `content` that must be a UTF-8 string of maximum 280 characters.
 
-Now we can go back to the sandbox and call that function. You might need to sign out of the sandbox first and reauthenticate.
+Inside the function, we use [`let`](../reference/functions.md#let) to create a local variable. This is like declaring a variable inside a function in other languages, but with Clarity's unique syntax. We're creating a variable called `id` and setting it to the current message count plus 1.
 
-When we do, you'll see the Leather wallet pop up with a transaction confirmation. You'll also see a little notice at the top that no transfers besides fees will occur.
+Here's where Clarity might trip you up if you're coming from other languages. Notice how we write `(+ (var-get message-count) u1)` instead of something like `message-count + 1`. In Clarity, operators like `+`, `-`, `>`, and `<` are actually functions that use prefix notation. So `(+ 2 3)` means "add 2 and 3" (instead of `2 + 3` like you'd write in JavaScript or Python). This is part of Clarity's LISP-inspired syntax where everything is a function call.
```

Comment:
```suggestion
*Here's where Clarity might trip you up if you're coming from other languages.* Notice how we write `(+ (var-get message-count) u1)` instead of something like `message-count + 1`. In Clarity, operators like `+`, `-`, `>`, and `<` are actually functions that use prefix notation. So `(+ 2 3)` means "add 2 and 3" (instead of `2 + 3` like you'd write in JavaScript or Python). This is part of Clarity's LISP-inspired syntax where everything is a function call.
```

seems important, recommend italics for the qualifier